x509/common.go

```diff
@@ -20,3 +21,26 @@ func readFile(filename string) string {
 	subcommands.DieNotNil(err)
 	return string(data)
 }
+
+type HsmInfo struct {
+	Module     string
+	Pin        string
+	TokenLabel string
+}
+
+type fileStorage struct {
+	Filename string
+}
+
+type hsmStorage struct {
+	HsmInfo
+	Label string
+}
+
+var factoryCaKeyStorage KeyStorage = &fileStorage{FactoryCaKeyFile}
+
+func InitHsm(hsm HsmInfo) {
+	if hsm.Module == "" {
```

Comment:
this confuses me. we only call InitHSM if `len(hsmModule) > 0`. So how would this get set?

Reply:
This was a bug, introduced while I shuffled the code between different options to implement this.

Fixed: https://github.com/foundriesio/fioctl/pull/278/commits/eb88423dfca9700fdff6102dc2b0fbbccb61b389